crates/miden-objects/src/note/note_tag.rs

```diff
@@ -440,7 +494,7 @@ mod tests {
         let private_account_id = AccountId::try_from(PRIVATE_ACCOUNT_INT).unwrap();
 
         // Expected private tag with LOCAL_EXECUTION_WITH_ALL_NOTE_TYPES_ALLOWED.
-        let expected_private_local_tag = NoteTag(0b11110011_00010101_00000000_00000000);
+        let expected_private_local_tag = NoteTag::from(0b10110011_00010101_00000000_00000000);
```

Comment:
For all of these tests, we're specifying the expected tag using the `0b...`, so we can be sure that whatever is returned matches that exactly. For that reason, I would avoid using `NoteTag::from` here and instead compare exactly against that value, e.g.:

```rust
let expected_private_local_tag = 0b10110011_00010101_00000000_00000000u32;
// ...
assert_eq!(
    NoteTag::from_account_id(private_account_id, NoteExecutionMode::Local).unwrap().as_u32(),
    expected_private_local_tag,
);
```

I would do this for all of the tests where you've changed it from the previous `NoteTag(expected_tag)` to `NoteTag::from(expected_tag)`.